code/network/multi_endgame.cpp

```diff
@@ -322,6 +323,10 @@ void multi_endgame_cleanup()
 {
 	int idx;
 
+	if (HUD_config.is_observer) {		// Do not forget restore players HUD
+		hud_config_restore();
```

Comment:
I think it would be better to use `hud_config_as_player` here since the opposing function `hud_config_as_observer` was used for creating the backup.

Reply:
I have an argument on this:
It was my initial train of thoughts to use `hud_config_as_player`. But it only do `hud_config_restore()` so it;s name could be misleading - since I can imagine situation in which factual _observer hud_ will be restored.

Reply:
The problem I see here is that the original `hud_config_backup` call was made through `hud_config_as_observer` so if you call `hud_config_restore` directly instead of `hud_config_as_player` you effectively rely on the internal implementation of `hud_config_as_observer` which is never a good idea.

Yes, all `hud_config_as_player` does is call a single function but if the operation is extended in the future then the places which call `hud_config_restore` now may be missed or would need to be updated.

Reply:
What shell we do about `hud_config_as_player` is plain misleading then?

Reply:
Why is it misleading? It's the opposite to `hud_config_as_observer` so it does what the name says...

Reply:
As I said earlier: what it actually do not necessarily sets hud config as player. It could set it as observer same way.

Reply:
Then maybe `hud_config_as_player` should check `HUD_config.is_observer` before calling `hud_config_restore`? Then it would only restore the HUD when the observer mode was active before.

Reply:
Yes! How come I couldn't figure it myself before :) It is all because of that smeared state context.
Will do.

Reply:
Welcome to the world of game development where every bit of game state is scattered over three different files...

Reply:
While it could be like that. How I miss those perfectly atomic states.

![observer](https://user-images.githubusercontent.com/21180686/46916798-32f31380-cfc8-11e8-8878-ed611504eaa4.png)